beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -58,9 +59,11 @@ type Node struct {
 	unrealizedFinalizedEpoch primitives.Epoch             // the epoch that would be finalized if the block would be advanced to the next epoch.
 	balance                  uint64                       // the balance that voted for this node directly
 	weight                   uint64                       // weight of this node: the total balance including children
+	weightWithoutBoost       uint64                       // weight of this node from attestations only - no proposer boost score
```

Comment:
This seems like a mistake, why do you need to add the boost to every single node instead of just checking that the node is an ancestor of the boosted root?